server/pkg/key/key.go

```diff
@@ -32,3 +32,9 @@ func (k Key) IsValid() bool {
 func (k Key) String() string {
 	return k.key
 }
+
+func (k Key) Clone() Key {
+	return Key{
+		key: k.key,
+	}
+}
```

Comment:
actually Key.Clone is not needed because it does not include any pointers as a field